### DISCUSSION TITLE
BirdWeather Posting Some but Not All Birds Reported

### DISCUSSION THREAD

**Discussion by dalogue1** [Nachtzuster/BirdNET-Pi#126] [General]
When my station detected a "Scaly-breasted Munia" and reported it to BirdWeather, I got the following response in my BirdNET-Pi log and the Munia was not in my species detected on BirdWeather.  So I do not know if BirdWeather is blocking "highly unlikely" birds or why it is not shown.

The following is where the message from [utils.reporting] originates so no exception was thrown. Just a 201 response_status_code which I think means success.

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/06aecb3d-2b1d-4532-af34-642697113b37)
 

> 15:52:17---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
> 15:52:17---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
> 15:52:17---[server][INFO] 9.0;12.0-('Lonchura punctulata_Scaly-breasted Munia', 0.8400175)
> 15:52:17---[server][INFO] 12.0;15.0-('Lonchura punctulata_Scaly-breasted Munia', 0.62650156)
> 15:52:17---[birdnet_analysis][INFO] 2024-06-26;15:52:01;Lonchura punctulata;Scaly-breasted Munia;0.84;29.7349;-95.4698;0.7;26;1.25;0.0;Scaly-breasted_Munia-84-2024-06-26-birdnet-15:52:01.mp3
> **15:52:24---[utils.reporting][INFO] Soundscape POST Response Status - 201**
> **15:52:25---[utils.reporting][INFO] Detection POST Response Status - 201**
> 15:52:31---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-26-birdnet-15:52:16.wav
> 15:52:31---[server][INFO] READING AUDIO DATA...
> 15:52:31---[server][INFO] READING DONE! READ 5 CHUNKS.
> 15:52

**Comment by dalogue1** [comment]
Here is another instance of it posting but not appearing on BirdWeather:

> 7:44:17---[server][INFO] ANALYZING AUDIO...
> 17:44:17---[server][INFO] DONE! Time 0.51 SECONDS
> 17:44:17---[server][INFO] 0.0;3.0-('Coccothraustes vespertinus_Evening Grosbeak', 0.23129572)
> 17:44:17---[server][INFO] 3.0;6.0-('Turdus philomelos_Song Thrush', 0.051936727)
> 17:44:17---[server][INFO] 6.0;9.0-('Lonchura punctulata_Scaly-breasted Munia', 0.38253924)
> 17:44:17---[server][INFO] 9.0;12.0-('Lonchura punctulata_Scaly-breasted Munia', 0.8281633)
> 17:44:17---[server][INFO] 12.0;15.0-('Lonchura punctulata_Scaly-breasted Munia', 0.59557533)
> 17:44:17---[birdnet_analysis][INFO] 2024-06-26;17:44:01;Lonchura punctulata;Scaly-breasted Munia;0.8282;29.7349;-95.4698;0.7;26;1.25;0.0;Scaly-breasted_Munia-83-2024-06-26-birdnet-17:44:01.mp3
> 17:44:20---[utils.reporting][INFO] Soundscape POST Response Status - 201
> 17:44:20---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:44:31---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-26-birdnet-17:44:16.wav
> 17:44:31---[server][INFO] READING AUDIO DATA...

**Comment by dalogue1** [comment]
This is what it does with a Blue Jay that just happened.  The Blue Jay appeared just fine on BirdWeather.

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/fed96609-02d4-485b-96ae-ff663dd8743b)


> 17:48:32---[server][INFO] ANALYZING AUDIO...
> 17:48:32---[server][INFO] DONE! Time 0.52 SECONDS
> 17:48:32---[server][INFO] 0.0;3.0-('Cyanocitta cristata_Blue Jay', 0.95974433)
> 17:48:32---[server][INFO] 3.0;6.0-('Cyanocitta cristata_Blue Jay', 0.97177595)
> 17:48:32---[server][INFO] 6.0;9.0-('Cyanocitta cristata_Blue Jay', 0.9508639)
> 17:48:32---[server][INFO] 9.0;12.0-('Cyanocitta cristata_Blue Jay', 0.9469712)
> 17:48:32---[server][INFO] 12.0;15.0-('Cyanocitta cristata_Blue Jay', 0.9661588)
> 17:48:32---[birdnet_analysis][INFO] 2024-06-26;17:48:17;Cyanocitta cristata;Blue Jay;0.9597;29.7349;-95.4698;0.7;26;1.25;0.0;Blue_Jay-96-2024-06-26-birdnet-17:48:17.mp3
> 17:48:32---[birdnet_analysis][INFO] 2024-06-26;17:48:17;Cyanocitta cristata;Blue Jay;0.9718;29.7349;-95.4698;0.7;26;1.25;0.0;Blue_Jay-97-2024-06-26-birdnet-17:48:17.mp3
> 17:48:33---[birdnet_analysis][INFO] 2024-06-26;17:48:17;Cyanocitta cristata;Blue Jay;0.9509;29.7349;-95.4698;0.7;26;1.25;0.0;Blue_Jay-95-2024-06-26-birdnet-17:48:17.mp3
> 17:48:33---[utils.reporting][WARNING] Extraction exists. Moving on: BirdSongs/Extracted/By_Date/2024-06-26/Blue_Jay/Blue_Jay-95-2024-06-26-birdnet-17:48:17.mp3
> 17:48:33---[birdnet_analysis][INFO] 2024-06-26;17:48:17;Cyanocitta cristata;Blue Jay;0.947;29.7349;-95.4698;0.7;26;1.25;0.0;Blue_Jay-95-2024-06-26-birdnet-17:48:17.mp3
> 17:48:33---[utils.reporting][WARNING] Extraction exists. Moving on: BirdSongs/Extracted/By_Date/2024-06-26/Blue_Jay/Blue_Jay-97-2024-06-26-birdnet-17:48:17.mp3
> 17:48:33---[birdnet_analysis][INFO] 2024-06-26;17:48:17;Cyanocitta cristata;Blue Jay;0.9662;29.7349;-95.4698;0.7;26;1.25;0.0;Blue_Jay-97-2024-06-26-birdnet-17:48:17.mp3
> 17:48:35---[utils.reporting][INFO] Soundscape POST Response Status - 201
> 17:48:36---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:48:36---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:48:37---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:48:37---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:48:38---[utils.reporting][INFO] Detection POST Response Status - 201
> 17:48:47---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-26-birdnet-17:48:32.wav
> 17:48:47---[server][INFO] READING AUDIO DATA...

**Comment by dalogue1** [comment]
I verified that the bird is on the BirdWeather species list.

Anyone have any ideas?

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/b41dbbab-65a3-4c66-8830-bf9697d58fc2)

**Reply by Frono** [reply]
I have a Birdnet-Pi and a Birdweather backed PUC. The microphones for the Pi and the PUC sit directly next to each other. Watching BirdWeather I notice large discrepancies in counts between the Pi and the PUC via the Birdweather site. Strangly enough it seems to vary by bird. Hits on Northern Cardinal are within 5%, carolina wren is 150% difference, house finch show a roughly 400% difference (both are within the last 24 hours). I'm thinking this may be down to frequency response/gain/filtering differences. But am still trying to sus out what I can do to maximize the Pi's performance...

**Reply by dalogue1** [reply]
1. Is the BirdNET-Pi producing more/fewer detections than the PUC?
2. Is the BirdWeather site doing any type of analysis at all or is it just logging the detections?
3. On a RPi 5 with NVMe, there is excess performance and I was wondering if there are any adjustments to settings to improve the detections.

**Reply by Frono** [reply]
1. Inconsistent. Some birds/times Pi is higher, sometimes lower...
2. No idea. I think it just logs for the Pi, but as far as I know the PUC is a black box affair. The only output I can see from it is on Birdweather. So it may well be modifying the output...
3. The only adjustments I know of for sound other than physical (mic selection, mic placement, etc...) are the alsa utilities to alter the microphone level driver gain, and the sigmoid adjustment, and confidence settings within the BirdNetPi Tools/Setting/Advanced_settings. There are positive/false-positive balancing effects via combinations of the model/range model/occurrence frequency in Tools/Setting/Basic_Settings...
Someone let me know if I missed anything.
I wrestle with trying to adjust the physical and microphone settings. I'd love to hear anyone elses input on a procedural best practice for just something as simple as deciding how to set microphone input gain vs. sigmoid...

**Reply by dalogue1** [reply]
I think a simple feature to detect birds as normal but only to report a bird to BirdWeather when more than one of the same type of bird has been detected within the last 24 hours.  That would avoid detections like the following on my station without adding the birds to the exclusion list. These are all just one off mis-detections! (I have listened to the sounds)

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/4885fe1f-f133-4785-a6d1-16ec5a25387e)

**Reply by alexbelgium** [reply]
Hi,

"On a RPi 5 with NVMe, there is excess performance and I was wondering if there are any adjustments to settings to improve the detections." -> you can change the "overlap" factor to have more precise analysis at the expense of more cpu. An overlap of 0.5s means that instead of analyzing 0-3 ; 3-6 ; 6-9 ; 9-12 ; 12-15s in a 15s sequence it will analyse : 0-3 ; 2.5-5.5 ; 5.0-8.0 ; 7.5-10.5 ; 10-13 ; 12.5-15.5s (anything beyond 15s is just blank if this is the duration of the registered clip). This will therefore allow you to do 20% more anlaysis on this 15s segment.

Other than that, device performance has no impact. @Frono mentioned all the critical elements. What I am doing is that I've installed homeassistant with Birdnet-pi as addon which allows to run 2 instances in parallel on the same device, with the exact same parameters except for a single difference that you want to investigate.

This is how I tested the different microphone here for example : SNR on 6 microphones https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-9706951 ; test of EM272 vs hyperx quadcast here : https://github.com/mcguirepr89/BirdNET-Pi/discussions/1092

Clearly, upgrading both my microphone and my aux/usb converter led to much higher detection rates ; although the rtsp feed (with camera noise canceling enabled) was already quite good

**Reply by dalogue1** [reply]
I changed my overlap from 0s to 1s to help with the analysis and the analysis time changed from 0.5 sec to 0,67 sec which is not a problem.  I will see how that goes.

Another possibility is to get a wireless lavalier microphone and place it as far away from the current microphone as possible.  I could then run 2 instances of the BirdNET-Pi as you described that are capturing similar data feeds.

**Reply by alexbelgium** [reply]
For a remote mic, I'm using a system described to me by @lloydbayley : you can use a couple of "USB to ethernet" adapters to use an ethernet cable as USB extension cord then connect the USB mic on the other end! It's great to push a USB mic further than the rpi itself (to better position it, and avoid interferences). It works great

**Reply by dalogue1** [reply]
How does a couple of "USB to Ethernet" adapters work.  I understand USB converts to ethernet but on the other end of the ethernet cable the adapter has a male USB connector.  How would you connect the USB microphone to that?


Do you mean one of these?
https://www.amazon.com/Siawclub-Ethernet-Compatible-Computers-Keyboards/dp/B0CTGTH26N/ref=asc_df_B0CTGTH26N/?tag=hyprod-20&linkCode=df0&hvadid=692875362841&hvpos=&hvnetw=g&hvrand=2381466721834077910&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9028158&hvtargid=pla-2281435179498&psc=1&mcid=9f750ba3af893ce1841fea1ae2128d88&hvocijid=2381466721834077910-B0CTGTH26N-&hvexpln=73&gad_source=1

It is funny that one set of these is $9 and two sets is $10.
https://www.amazon.com/dp/B08FMLGNQN/ref=vp_d_vp_dp_d_dex_hr_v3_t1_pd?_encoding=UTF8&pf_rd_p=78bd751c-afcc-4a5e-987d-d425671f7a10&pf_rd_r=TYBWPYPATD226D5CAS37&pd_rd_wg=h3YM3&pd_rd_i=B08FMLGNQN&pd_rd_w=FPEai&content-id=amzn1.sym.78bd751c-afcc-4a5e-987d-d425671f7a10&pd_rd_r=e280ac15-8eb0-4d39-b8d5-4cc21578aebb&psc=1


I was considering a microphone extension cable made from a coaxial cable.

**Reply by lloydbayley** [reply]
One has a female USB :)

**Reply by dalogue1** [reply]
But "normal" USB to Ethernet adapters are generally male USB.  This is a special set of adapters made for this purpose.

Very nice!  I never knew these existed.

**Reply by lloydbayley** [reply]
I suppose a better description is USB Extenders via UTP.
https://www.amazon.com.au/gp/product/B0029HRYBK/

**Comment by alexbelgium** [comment]
"Detection POST Response Status - 201" means that it was posted successfully to birdweather so from an app side it should work... I know the owner of birdweather goes sometimes on the repo so hopefully he can help you

**Comment by dalogue1** [comment]
The Scaly-breasted Munia is not present in the Species Probability dialog for my location on their website.

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/ea703033-3de8-4dca-85b5-5a055d3fd477)

**Comment by dalogue1** [comment]
I sent an email to Tim the Founder of BirdWeather and he adjusted the Species Probability for that specific bird on my station #5355 and now the Scaly-breasted Munias are being logged.

Very exciting!

Thanks again, Tim.

![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/133064876/fe6a5ee9-a497-401f-b8e3-89e8b31295a8)

**Reply by dalepa** [reply]
I have about 20 scaly-breated too. I was able to white list them and get them to show in my birdnetpi. But I don't see them in birdweather...

Do I need to email him too?

**Reply by dalogue1** [reply]
Yes and tell him your station id number.

**Reply by dalogue1** [reply]
I am curious about where you are seeing Scaly-Breasted Munias as well.  What city?  You could just give the station number as well.

BTW this is what I said in my email to tim@BirdWeather.com
"have a BirdNET-Pi station #5355 reporting to BirdWeather. I have no less than 50 detections of Scaly-breasted Munias on my system a day. There is no question"

His response was "Done."

**Comment by dalepa** [comment]
I have at least 20 Scaly-Breasted Munias that's have been around yearly
since 2021.  #11868
Cypress Texas

**Reply by dalogue1** [reply]
Is your station reporting to BirdWeather yet?  I don't see any readings after 3/13/2025.  

Sometimes the BirdNetPi is working but is not connected to the internet.  See the previous thread: https://github.com/Nachtzuster/BirdNET-Pi/discussions/218

**Comment by dalepa** [comment]
This one isn't at the moment.  It will be back up this week.

**Reply by dalogue1** [reply]
I see your Scaly-Breasted Munia showed up.  (You can see mine below yours.)
![image](https://github.com/user-attachments/assets/dd07fbdf-69eb-4b2d-8aa9-9f766710e0ed)
